studio/queues/rabbit_queue.py

```diff
@@ -98,9 +102,9 @@ def connect(self):
         params = pika.URLParameters(self._url)
         new_connection = pika.SelectConnection(
             params,
-            on_open_callback=self.on_connection_open,
-            on_open_error_callback=None,
-            on_close_callback=self.on_connection_closed,
+            on_open_callback=self._on_connection_open,
+            on_open_error_callback=self._on_connection_error,
+            on_close_callback=self._on_connection_closed,
             custom_ioloop=None,
```

Comment:
This is the main part of the fix - we should register "connection error" callback, not only "connection done" callback.